clang/lib/CIR/Dialect/Transforms/FlattenCFG.cpp

```diff
@@ -135,13 +135,10 @@ class CIRScopeOpFlattening : public mlir::OpRewritePattern<cir::ScopeOp> {
     // Split the current block before the ScopeOp to create the inlining
     // point.
     auto *currentBlock = rewriter.getInsertionBlock();
-    auto *remainingOpsBlock =
+    mlir::Block *continueBlock =
         rewriter.splitBlock(currentBlock, rewriter.getInsertionPoint());
-    mlir::Block *continueBlock;
-    if (scopeOp.getNumResults() == 0)
-      continueBlock = remainingOpsBlock;
-    else
-      llvm_unreachable("NYI");
+    if (scopeOp.getNumResults() > 0)
+      continueBlock->addArguments(scopeOp.getResultTypes(), loc);
```

Comment:
I'm a bit puzzled on how this is working in practice. Usually a `addArguments` on one side is accompanied by a `getArgument` on the side of the code that handles the target block, and since I don't see the equivalent here I'm lost - seems like the tests are checking for the right thing though. Can you elaborate?

Reply:
ah cool, so it already had the support even though we were not using it, thanks!